src/translate_c.zig

```diff
@@ -1127,6 +1127,86 @@ fn transOffsetOfExpr(
     return fail(c, error.UnsupportedTranslation, expr.getBeginLoc(), "TODO: implement complex OffsetOfExpr translation", .{});
 }
 
+/// Translate an arithmetic expression with a pointer operand and a signed-integer operand.
+/// Zig requires pointer arithmetic integer operands to be unsigned so we need to check the value at runtime
+/// and add or subtract the unsigned absolute value.
+/// Zig pointer arithmetic is also not commutative (unlike C); the pointer operand needs to be on the left.
+/// The + and - operators in C are not sequence points so it should be safe to switch the order.
+/// We store the lhs and rhs expression results into temporary variables to prevent double-evaluation
+/// of side-effects.
+///
+/// Given a pointer expression `ptr` and signed int expression `idx` we translate as follows:
+/// `ptr + idx` and `idx + ptr` become:
```

Comment:
You can also cast the signed int to a `usize` with `@bitCast(usize, @intCast(isize, idx))` to tidy up the generated code.

Reply:
Wouldn't that fail if `idx` is negative? Having that said that I probably should do the `absCast` in the temp variable assignment instead of both arms of the `if`

Reply:
Nope, the wrap-around handles that for you.

Reply:
Well that certainly makes the code cleaner. Updated.